### ISSUE TITLE
Choose the right subscription type

### ISSUE THREAD

**Issue by NicolasReyDotNet** [EventStore/EventStore#2402]
Hi,

I'm completely new to Event Sourcing and to EventStore, and I try hard to learn how it works, so I recently read the book from @alexeyzimarev  ([Github repo](https://github.com/PacktPublishing/Hands-On-Domain-Driven-Design-with-.NET-Core)), he deals with Event sourcing and used EventStore in his demo, and he explained how to handle projections from EventStore using **Catch-up subscriptions** to construct read models. In the final chapter (13) Alexey told that for the read model projection, this subscription mode leads us to encounter the "Competiting consumer" problem on horizontal scaling of client subscribers, which is also explained in [your doc](https://eventstore.com/docs/getting-started/reading-subscribing-events/index.html?tabs=tabid-6%2Ctabid-dotnet-read-event%2Ctabid-create-sub-dotnet#persistent-subscriptions). I fully understand the problem, especially in a Kubernetes+Docker environment , and that switching to **Persistent subscriptions** solves this problem.

Well I have a very trivial question, but I can't easily find an answer  : what does the **Catch-up subscriptions** allows to do that the **Persistent subscriptions** does not ? And when should I use the **Catch-up subscriptions** instead of the **Persistent subscriptions** if it's _better_ ? 
Especially if I want to replay all the events, as I understood in the **Persistent subscriptions**  we don't manually handle the **position**, the server does it for us, so is it possible to replay the event as we should do in the **Catch-up subscriptions**  droping the **position** from our DB ?

Thank you for your help, any sample would be really appreciated

**Comment by alexeyzimarev** [comment]
Catch-up (also known as "volatile") subscriptions have a single subscriber, which is a function. It is initiated by the client and there's nothing on the server-side that is left of such a subscription when it closes. Catch-up subscriptions are very useful for building read-models when the order of events is essential for the projection. They also can subscribe to the `$all` stream. The client controls the checkpoint (stream offset), and I advise in the book to keep the checkpoint close to the read-model itself (same storage, same DB, etc.).

Persistent subscriptions are established _on the server_. You can connect as many clients as you want to it, and the server will take care of load distribution, according to the policy you specify. The server will also take care of storing the checkpoint. You can also decide to ACK/NACK (or park) messages, and the server will keep track of messages that you parked or NACKed. But, as the docs mentioned, the order is not guaranteed for persistent subscriptions, for a simple reason of competing consumers. Also, at the current client version, you cannot create a persistent subscription for the `$all` stream.

**Comment by NicolasReyDotNet** [comment]
Thanks, its clear

**Comment by NicolasReyDotNet** [comment]
Thanks it's clear now.

@alexeyzimarev : could you confirm then that for the application responsible for the projections based on catch-up, we must ensure that only one instance is running at once ? (to avoid the"Competiting consumer")

**Comment by alexeyzimarev** [comment]
If you are using a catch-up subscription, it is impossible to get competing consumers if you run several instanced of your application in parallel. Each instance will get all the events. As I wrote, the server doesn't care in this case.

**Comment by NicolasReyDotNet** [comment]
@alexeyzimarev 
Ok right, I understand.

So because having more than one instance dealing with projection is not desirable (we will apply more than one time the same event, update a field for example, to the read model, one per instance / subscription), it's the expected behavior to limit just one instance of the application dealing with the projection (catch-up subscription based) ?

**Comment by alexeyzimarev** [comment]
Catch-up subscriptions are initiated by the client. So, each subscription is considered as a separate subscription. The server doesn't care. Each subscription gets all events, so you cannot load-balance subscribers, at least the server won't do it for you.

**Comment by NicolasReyDotNet** [comment]
@alexeyzimarev 
ok it makes sense, but just to fully understand with no doubt :
- I create a dedicated application (ReadModelProjection as example) in charge of construction of all read models that starts the projection catch up subscription and constucts each of them from the $all stream from the last stored checkpoint, as you do in your book
- suppose this application is in a docker container in a Kubernetes cluster, if I do not restrict the number of its running instance to only one, each of these instances will handle the same events from the server (which doesn't care the number of subscribers same instance or not) and each instance will try to make the same actions in parallel, lets say "update a name field to "new name" for a specific aggregate id=123" on the read model DB. And we do want such behavior.

I'm a right or am I missing something ?

**Comment by alexeyzimarev** [comment]
What you wrote is correct.

I wouldn't do a single service that handles all the read models though.

**Comment by NicolasReyDotNet** [comment]
@alexeyzimarev 

> I wouldn't do a single service that handles all the read models though.

Ok interesting! What approch would you have then ? An application per Read Model "context" ? Or do not split the application responsible for the projection from the application handling commands and appending event to the event store ? What are the cons of use one application for all read models projection ?

**Comment by alexeyzimarev** [comment]
It's an interesting discussion but I don't think it belongs here. I can suggest for you to join the DDD/CQRS/ES Slack org, there are a lot of people there who practice Event Sourcing in production and are happy to share their experience https://j.mp/ddd-es-cqrs

**Comment by NicolasReyDotNet** [comment]
@alexeyzimarev 
You are right, thank you for the link.
May I ask you a last point : I opened an Issue on your Book [Github](https://github.com/PacktPublishing/Hands-On-Domain-Driven-Design-with-.NET-Core/issues/15), could you have a check ?
